spec/models/user_spec.rb

```diff
@@ -2,8 +2,19 @@
 
 RSpec.describe User, type: :model do
   it 'knows its role in society' do
-    user = User.create
+    user = User.create! given_name: 'Dennis', family_name: 'Ritchie', email: 'dmr@bell-labs.com'
     user.roles.create role: :admin # Make the user an admin.
     expect(user).to be_admin # RSpec for "assert that user.admin? is true"
   end
+
+  it 'knows its name' do
+    user = User.new given_name: 'Alan', family_name: 'Turing'
+    expect(user.name).to eq 'Alan Turing'
+
+    user = User.new given_name: '', family_name: 'Fureigh'
+    expect(user.name).to eq 'Fureigh'
+
+    user = User.new given_name: 'John', family_name: 'Hall', preferred_name: 'Mad Dog'
+    expect(user.name).to eq 'Mad Dog Hall'
+  end
```

Comment:
I think RSpec convention would prefer something like this (as opposed to multiple independent "tests" in one `it`):
```rb
describe '#name' do
  it 'handles normal first/last names' do
    user = User.new given_name: 'Alan', family_name: 'Turing'
    expect(user.name).to eq 'Alan Turing'
  end
  it 'handles people with single names'
    user = User.new given_name: '', family_name: 'Fureigh'
    expect(user.name).to eq 'Fureigh'
  end
  it 'handles people with preferred names'
     user = User.new given_name: 'John', family_name: 'Hall', preferred_name: 'Mad Dog'
     expect(user.name).to eq 'Mad Dog Hall'
  end
end
```